datafusion/core/src/datasource/listing/url.rs

```diff
@@ -234,7 +239,7 @@ impl ListingTableUrl {
     }
 
     /// List all files identified by this [`ListingTableUrl`] for the provided `file_extension`
-    pub(crate) async fn list_all_files<'a>(
+    pub async fn list_all_files<'a>(
```

Comment:
is it related to the PR?

Reply:
I'm happy to separate this out if preferred. I made this change because it allows easier debugging for future users who are unsure which files the `ListingTableUrl` will actually list. This means one can see whether the issue is with the url itself, or another issue